### ISSUE TITLE
False positive on `ReusedModifierInstance` when shadowing modifier

### ISSUE THREAD

**Issue by BraisGabin** [appKODE/detekt-rules-compose#30]
After update to 1.3.0 fixed a lot a false positives of `ReusedModifierInstance`. But there is one that it wasn't fixed yet. I created a test case to make it easy to reproduce it:

```kotlin
@Composable
fun Extra(a: @Composable (Modifier) -> Unit) {}
@Composable
fun Test(
  modifier: Modifier = Modifier, 
) {
  Column(modifier) {
    Extra { modifier ->
      Text("Hi", modifier)
    }
  }
}
```

The `modifier` that I'm using on `Text` is not the same that the one defined on `Test`.

**Comment by dimsuz** [comment]
Thank you, I'll take a look!
